src/main/java/de/tum/in/www1/artemis/domain/User.java

```diff
@@ -240,6 +244,17 @@ public void setPersistentTokens(Set<PersistentToken> persistentTokens) {
         this.persistentTokens = persistentTokens;
     }
 
+    public GuidedTourSettings getGuidedTourSettings() {
+        if (this.guidedTourSettings == null) {
```

Comment:
I would suggest to set `guidedTourSettings` by default for each user. Here inside `@Column` definition you can specify value by setting `columnDefinition` variable. This will work for the new users. For old users you will need to use  `addNotNullConstraint` in Liquibase changelog. You can check how I used it recently for `complaintType` in one of the previous changelogs and more detailed description here https://www.liquibase.org/documentation/changes/add_not_null_constraint.html

Reply:
@maxr96 thank you for the recommendation. But I think (at least for now) it is better to keep the configuration like this. The settings JSON is an object to change (since this is the first draft) and with the default value I would have to update every single user for every update on the settings.